### DISCUSSION TITLE
Building docker image and pushing it to azure container registry or azure container app

### DISCUSSION THREAD

**Discussion by MasterThreadInf** [unoplatform/uno#10471] [Q&A]
In visual studio 2022, currently I don't see docker support for Uno application.  Is it possible to build and push docker image to azure container app or container registry ?  
If it not possible then ,is there any other way where I don't have to use github and I can directly commit my code from VS to azure container app or container registry?

**Comment by jeromelaban** [comment]
At this time, we do not have deocumentation on how to do this, but since Uno Platform WebAssemby apps are a set of static files, any container image which can serve static files can be used.

If you find a way to build an app this way, let us know here so we can enhance our documentation!

**Reply by MasterThreadInf** [reply]
Thanks for reply. Ok , but is there any way to enable docker support for uno app?
Is there any way where I can use azure authentification for Azure static web app ? Also what are custom login(authentification) options in uno ? Unfortunately, there isn't any sample solution for authentification.

**Reply by jeromelaban** [reply]
As I mentioned, docker support is not documented and an Uno app is a set of static files so there's nothing specific in the app to change about it.

For authentication, [uno supports MSAL](https://platform.uno/docs/articles/interop/MSAL.html).

**Reply by MasterThreadInf** [reply]
Yes I understood its not documented ,but I asked is there any way to enable it so that it opens doors to many things in azure like authentification.  Thanks for the link ,but is there any way for custom authentification ?

**Reply by jeromelaban** [reply]
Custom authentication would mean some of your code, so it's up to you to build it. What kind of authentication do you want to build? Do you have an existing example using another technology?

**Reply by MasterThreadInf** [reply]
I'm talking about this https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Tutorials/Authentication/HowTo-Authentication.html . In there custom login is used without any provider. I followed all the step but getting CS0006 error .dll not found. 

sample : https://github.com/MasterThreadInf/UnoAuthentificationDemo

**Reply by jeromelaban** [reply]
I see. @nickrandolph may be able to help.

**Reply by nickrandolph** [reply]
@MasterThreadInf  I'm not sure what you're asking for here and I can't see in your sample where you're adding authentication?

**Reply by MasterThreadInf** [reply]
@nickrandolph  If you follow this path(https://github.com/MasterThreadInf/UnoAuthentificationDemo/blob/main/authen/authen.Shared/MainPage.xaml.cs) there you'll find that I've added authetification. If you try the solution to run then you'll understand what I'm asking. I followed the instruction from  https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Tutorials/Authentication/HowTo-Authentication.html.
